presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1037,57 +1053,35 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
 
             Optional<ViewDefinition> optionalView = metadata.getView(session, name);
             if (optionalView.isPresent()) {
-                Statement statement = analysis.getStatement();
-                if (statement instanceof CreateView) {
-                    CreateView viewStatement = (CreateView) statement;
-                    QualifiedObjectName viewNameFromStatement = createQualifiedObjectName(session, viewStatement, viewStatement.getName());
-                    if (viewStatement.isReplace() && viewNameFromStatement.equals(name)) {
-                        throw new SemanticException(VIEW_IS_RECURSIVE, table, "Statement would create a recursive view");
-                    }
-                }
-                if (analysis.hasTableInView(table)) {
-                    throw new SemanticException(VIEW_IS_RECURSIVE, table, "View is recursive");
-                }
-                ViewDefinition view = optionalView.get();
-
-                Query query = parseView(view.getOriginalSql(), name, table);
+                return processView(table, scope, name, optionalView);
+            }
 
-                analysis.registerNamedQuery(table, query);
+            Optional<ConnectorMaterializedViewDefinition> optionalMaterializedView = metadata.getMaterializedView(session, name);
 
-                analysis.registerTableForView(table);
-                RelationType descriptor = analyzeView(query, name, view.getCatalog(), view.getSchema(), view.getOwner(), table);
-                analysis.unregisterTableForView();
+            //TODO: More validation condition?
+            if (optionalMaterializedView.isPresent()) {
+                Statement statement = analysis.getStatement();
 
-                if (isViewStale(view.getColumns(), descriptor.getVisibleFields())) {
-                    throw new SemanticException(VIEW_IS_STALE, table, "View '%s' is stale; it must be re-created", name);
+                final List<Class<? extends Statement>> statementInSupport = ImmutableList.of(
+                        Explain.class,
+                        Prepare.class,
+                        Query.class,
+                        ShowStats.class);
+                if (statementInSupport.stream().noneMatch(support -> support.isInstance(statement))) {
+                    throw new SemanticException(NOT_SUPPORTED, table, "Materialized view is not supported in this statement.");
```

Comment:
In what situation we will need to exclude these statements?

Reply:
Not sure if I follow? This is inspired from views.

Reply:
I mean, why do we need to exclude Explain, Prepare, Query, and ShowStats?